Jenkinsfile

```diff
@@ -0,0 +1,169 @@
+import groovy.json.JsonOutput
+
+/** Map of Tox environments and associated capabilities */
+def environments = [
+  desktop: [
+    browserName: 'Firefox',
+    version: '47.0',
+    platform: 'Windows 10'
+  ],
+  mobile: [
+    browserName: 'Browser',
+    platformName: 'Android',
+    platformVersion: '5.0',
+    deviceName: 'Android Emulator',
+    appiumVersion: '1.5.3'
+  ]
+]
+
+/** Transform a map into a list of maps
+ *
+ * @param aMap map to transform
+ * @return a list of maps
+*/
+@NonCPS
+def entrySet(aMap) {
+  aMap.collect {
+    k, v -> [key: k, value: v]
+  }
+}
+
+/** Write capabilities to JSON file
+ *
+ * @param desiredCapabilities capabilities to include in the file
+*/
+def writeCapabilities(desiredCapabilities) {
+    def defaultCapabilities = [
+        build: env.BUILD_TAG,
+        public: 'public restricted'
+    ]
+    def capabilities = defaultCapabilities.clone()
+    capabilities.putAll(desiredCapabilities)
+    def json = JsonOutput.toJson([capabilities: capabilities])
+    writeFile file: 'capabilities.json', text: json
+}
+
+/** Run Tox
+ *
+ * @param environment test environment to run
+*/
+def runTox(environment) {
+  try {
+    wrap([$class: 'AnsiColorBuildWrapper']) {
+      withCredentials([[$class: 'StringBinding', credentialsId: 'SAUCELABS_API_KEY', variable: 'SAUCELABS_API_KEY']]) {
+        withCredentials([[$class: 'FileBinding', credentialsId: 'AMO_VARIABLES', variable: 'VARIABLES']]) {
+          withEnv(["PYTEST_ADDOPTS=--base-url=${PYTEST_BASE_URL} --driver=SauceLabs --variables=capabilities.json --variables=${VARIABLES} --color=yes"]) {
+            sh 'tox -e ' + environment
+          }
+        }
+      }
+    }
+  } catch(err) {
+    currentBuild.result = 'FAILURE'
+    throw err
+  } finally {
+    dir('results') {
+      stash environment
+    }
+  }
+}
+
+/** Send a notice to #fxtest-alerts on irc.mozilla.org with the build result
+ *
+ * @param result outcome of build
+*/
+def ircNotification(result) {
+  def nick = "fxtest${BUILD_NUMBER}"
```

Comment:
oh wow, that's clever.  so the nick will be fxtest + build num.  that way you could have multiple jobs posting simultaneously.  very nice.